k2/csrc/properties.cc

```diff
@@ -78,6 +79,43 @@ bool HasSelfLoops(const Fsa &fsa) {
   return false;
 }
 
+bool CheckCycles(StateId s, std::vector<bool> &visited, std::vector<bool> &back_arc,
+                 const std::vector<Arc> &arcs) {
+  if(visited[s] == false) {
+    visited[s] = true;
+    back_arc[s] = true;
+
+    std::unordered_set<StateId> adj;
+    for (const auto &arc : arcs)
+      if (arc.src_state == s)
+        adj.insert(arc.dest_state);
+
+    for (auto i = adj.begin(); i != adj.end(); ++i) {
+      if (!visited[*i] && CheckCycles(*i, visited, back_arc, arcs))
+        return true;
+      else if (back_arc[*i])
+        return true;
+    }
+  }
+  back_arc[s] = false;
+  return false;
+}
+
+// Detect cycles using DFS traversal
+bool IsAcyclic(const Fsa &fsa) {
+  StateId num_states = fsa.NumStates();
+  const std::vector<Arc> arcs = fsa.arcs;
+  std::vector<bool> visited(num_states, false);
+  std::vector<bool> back_arc(num_states, false);
+
+  for (StateId i = 0; i < num_states; i++)
+    if (CheckCycles(i, visited, back_arc, arcs))
+      return true;
+
+    return false;
```

Comment:
Better to add braces `{}` for the `for` loop.

Indentation does not work here.